### ISSUE TITLE
react/promise v3 doesn't work fine

### ISSUE THREAD

**Issue by roxblnfk** [temporalio/sdk-php#357]
In order to ensure asynchronous operation of user workflows, coroutines are implemented in the SDK, which use promises. It is a standard situation when we reject them with internal exceptions, which in some branches may not be handled.
However, in the paradigm of react/promise v3, this is considered undesirable practice. Therefore, when a rejected promise is completed, in which there is no exception handler, the [`error_log()`](https://www.php.net/manual/function.error-log) function is called with the error text.

The side effects of such behavior are unpredictable, and it is currently impossible to disable it. 

The issue about unhandled exceptions may be fixed once react/promise has the ability to [set a permanent handler](https://github.com/reactphp/promise/pull/249#discussion_r1345436865), not a one-time one.

---

Thanks to the tests in roadrunner-temporal, other deviations from the normal operation of the SDK were detected. These tests have been partially transferred to the SDK (see #359).

Therefore, I suggest temporarily abandoning react/promise v3 since version 2.6.1 until a solution is found.

**Comment by ruudk** [comment]
Is there an update on this? I noticed that I had to downgrade react/promise on my project in order to install this SDK. And now I see all kind of `Implicitly marking parameter $x as nullable is deprecated` errors on PHP 8.4 that are fixed in v3 with https://github.com/reactphp/promise/pull/260.

**Comment by roxblnfk** [comment]
React v3 still has several problems that prevent us from using it.

**Comment by ruudk** [comment]
I read the issue https://github.com/temporalio/sdk-php/pull/359 but saw some follow up PR's, one that introduced a global rejection handler: https://github.com/reactphp/promise?tab=readme-ov-file#set_rejection_handler

Isn't this what was missing?

**Comment by roxblnfk** [comment]
It's [one-time](https://github.com/reactphp/promise/blob/5f80055cc21ba7bcd3989e4902061fc12e2bcc1d/src/Internal/RejectedPromise.php#L38) handler

**Comment by ruudk** [comment]
Should there be a new issue to track this? I'm a bit surprised that the Temporal SDK has deprecations on PHP 8.4, which was released 6 months ago.

**Comment by roxblnfk** [comment]
Regarding deprecations in the SDK, definitely create an issue 👍 . However, deprecations in dependencies should be tracked by the dependencies themselves.

**Comment by ruudk** [comment]
Sure, but in this case, this SDK prevents installation of the latest major, so I think it would have been good if this project created that issue upstream. How else is it going to be fixed?

**Comment by roxblnfk** [comment]
Yes, it would be great to push the development of React/Promise. However, we still won't achieve the set goals, including those related to the `@yield` annotation.

Therefore, the most promising direction is to create a fork and maintain our own version of promises compatible with React. However, this is not a priority right now.
